graph/tree/tree.go

```diff
@@ -14,12 +14,13 @@ const (
 )
 
 type ServiceNode struct {
-	ID       string                 // unique identifier for the service node
-	Name     string                 // service name
-	Version  string                 // service version
-	Parent   *ServiceNode           // parent service node, or nil if no parent
-	Children []*ServiceNode         // children services nodes
-	Metadata map[string]interface{} // app-specific data
+	ID        string                 // unique identifier for the service node
+	Name      string                 // service name
+	Version   string                 // service version
+	Namespace string                 // Service namespace
```

Comment:
Again, not sure we need a convenience field here but it's not too bad given that it's only server-side.